quinn-proto/src/connection/mod.rs

```diff
@@ -75,6 +75,23 @@ where
     /// Exactly one prior to `self.rem_cids.offset` except during processing of certain
     /// NEW_CONNECTION_ID frames.
     rem_cid_seq: u64,
+    rem_cid_resettoken: Option<ResetToken>,
+
+    /// Cid rotation
+    ///
+    /// Expiration timestamp of local connection IDs
+    cids_timeline: VecDeque<CidTimeStamp>,
+    /// The lifetime of Local connection IDs
+    cids_timeout: Option<Duration>,
+    /// Number of local connection IDs that have been issued in NEW_CONNECTION_ID frames.
+    cids_issued: u64,
+    /// The sequence number of active (not yet rotated/retired by peer) local connection IDs
+    cids_active_seq: HashSet<u64>,
```

Comment:
I Since it's probably a tiny set one might have a fixed size array to manage those.